### ISSUE TITLE
sporadic test errors

### ISSUE THREAD

**Issue by evgeni** [FrostyX/tracer#149]
when building tracer on EL7, I've sporadically seen the following errors during tests:

```
+ cd tracer-0.7.3
+ nosetests .
.SSSSSEE..........SSSSSSSSSSS......SSSSSSSS....................SSSSS
======================================================================
ERROR: test_app_with_no_definition (tests.test_applications.TestApplications)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/builddir/build/BUILD/tracer-0.7.3/tests/test_applications.py", line 45, in test_app_with_no_definition
    self.assertEqual(app.type, Applications.DEFAULT_TYPE)
  File "/builddir/build/BUILD/tracer-0.7.3/tracer/resources/applications.py", line 202, in type
    elif self.is_session:
  File "/builddir/build/BUILD/tracer-0.7.3/tracer/resources/applications.py", line 196, in is_session
    return self.instances and self.instances[0].is_session
  File "/builddir/build/BUILD/tracer-0.7.3/tracer/resources/applications.py", line 261, in instances
    lambda process: self._attributes["name"] in [process.name(), process.real_name])
  File "/builddir/build/BUILD/tracer-0.7.3/tracer/resources/collections.py", line 90, in filtered
    if function(process):
  File "/builddir/build/BUILD/tracer-0.7.3/tracer/resources/applications.py", line 261, in <lambda>
    lambda process: self._attributes["name"] in [process.name(), process.real_name])
  File "/builddir/build/BUILD/tracer-0.7.3/tracer/resources/processes.py", line 82, in name
    return 'ssh-{0}-session'.format(re.split(' |@',' '.join(self._attr("cmdline")))[1])
IndexError: list index out of range
======================================================================
ERROR: test_application_processes (tests.test_applications.TestApplications)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/tracer-0.7.3/tests/test_applications.py", line 18, in test_application_processes
    self.assertIsInstance(application.instances, ProcessesCollection)
  File "/builddir/build/BUILD/tracer-0.7.3/tracer/resources/applications.py", line 261, in instances
    lambda process: self._attributes["name"] in [process.name(), process.real_name])
  File "/builddir/build/BUILD/tracer-0.7.3/tracer/resources/collections.py", line 90, in filtered
    if function(process):
  File "/builddir/build/BUILD/tracer-0.7.3/tracer/resources/applications.py", line 261, in <lambda>
    lambda process: self._attributes["name"] in [process.name(), process.real_name])
  File "/builddir/build/BUILD/tracer-0.7.3/tracer/resources/processes.py", line 82, in name
    return 'ssh-{0}-session'.format(re.split(' |@',' '.join(self._attr("cmdline")))[1])
IndexError: list index out of range
----------------------------------------------------------------------
Ran 68 tests in 0.251s
FAILED (SKIP=29, errors=2)
```

I *guess* this is related to the other processes running on the build nodes, but can't be sure.

**Comment by FrostyX** [comment]
Hello @evgeni, 
thank you for the report.

We already had this reported as #129 and hopefully, it should be fixed by PR #146. Version 0.7.4 is in bodhi now and should be in stable Fedora/EL soon.

**Comment by FrostyX** [comment]
As I said in the previous comment, I think this is already fixed.
